sd/etcd/client_test.go

```diff
@@ -0,0 +1,69 @@
+package etcd
+
+import (
+	"log"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+func TestNewClient(t *testing.T) {
+	ClientOptions := ClientOptions{
+		Cert:                    "",
+		Key:                     "",
+		CaCert:                  "",
+		DialTimeout:             (2 * time.Second),
+		DialKeepAline:           (2 * time.Second),
+		HeaderTimeoutPerRequest: (2 * time.Second)}
+
+	client, err := NewClient(
+		context.Background(),
+		[]string{"http://localhost:2379"},
```

Comment:
This is a be a basic unit test to test some of the NewClient function behavior. Etcd does not need to be running. Though it does register a _new client_, if you try to actually access any functions of that client it will return an `etcd cluster is unavailable or misconfigured` error.